### PR TITLE
Add migration test for ESPOS

### PR DESCRIPTION
**Pull request by chcao**
Automation for ESPOS migration. Register ESPOS and fully update the system, then remove ESPOS before migration, and then perform migration.

- Related ticket: https://progress.opensuse.org/issues/77272
- Verification run: 
https://openqa.nue.suse.com/tests/5004491#step/zypper_patch/9
https://openqa.nue.suse.com/tests/5004494#step/patch_sle/165
https://openqa.nue.suse.com/tests/5004509#step/zypper_patch/9
https://openqa.nue.suse.com/tests/5004501

### CODE REVIEW

File: lib/registration.pm [hunk 1]
```diff
@@ -415,6 +416,7 @@ sub process_scc_register_addons {
     #   tsm - Transactional Server Module
     #    we - Workstation
     #   wsm - Web and Scripting Module
+    # espos - ESPOS
```
Comment (lemon-suse) [inline lib/registration.pm:419]: Align and need give full expression for ESPOS.

File: tests/update/patch_sle.pm [hunk 2]
```diff
@@ -101,6 +101,17 @@ sub patching_sle {
         set_var('SKIP_INSTALLER_SCREEN', 1) if get_var('MAINT_TEST_REPO');
 
     }
+    #migration with ESPOS is not possible, remove it before upgrade
+    remove_espos;
+    if (get_var('FLAVOR', '') =~ /-(Updates|Incidents)$/ || get_var('KEEP_REGISTERED')) {
```
Comment (lemon-suse) [inline tests/update/patch_sle.pm:106]: It seems same as ltss, so do this once is ok.

**Comment by coolgw** [comment]
LGTM

**Comment by lemon-suse** [comment]
Please update the title and description, Migration testing of ESPOS -> Add migration test for ESPOS?  Automation for ESPOS migration. ->  Add automation test for ESPOS.

Besides,  please update the commit also and support detail commit message, merge the commits.

File: lib/registration.pm [hunk 3]
```diff
@@ -87,6 +87,7 @@ our %ADDONS_REGCODE = (
     'sle-live-patching'        => get_var('SCC_REGCODE_LIVE'),
     'SLES-LTSS'                => get_var('SCC_REGCODE_LTSS'),
     'SUSE-Linux-Enterprise-RT' => get_var('SCC_REGCODE_RT'),
+    ESPOS                      => get_var('SCC_REGCODE_ESPOS'),
```
Comment (lemon-suse) [inline lib/registration.pm:90]: ESPOS -> 'ESPOS'

Comment (chcao) [reply lib/registration.pm:90]: > ESPOS -> 'ESPOS'
If add single quotes, it would print errors when auto checking.

**Review by lemon-suse — APPROVED** [review]

**Pull request merged**